pkg/inspector/image-inspector.go

```diff
@@ -467,3 +414,153 @@ func createOutputDir(dirName string, tempName string) (string, error) {
 	}
 	return dirName, nil
 }
+
+// pullExtractAndInspectImage will use containers/image library to pull extract and get the image to be scanned
+// and will inspect it for its metadata.
+// Assumes the image to be pulled is with a transport prefix
+// or if not attempts to add the "docker://" prefix to the image.
+func (i *defaultImageInspector) pullExtractAndInspectImage() (*types.ImageInspectInfo, digest.Digest, error) {
+
+	policy, err := signature.NewPolicyFromBytes([]byte(DEFAULT_SIGN_POLICY))
+	if err != nil {
+		return nil, "", err
+	}
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return nil, "", err
+	}
+	defer policyContext.Destroy()
+
+	srcRef, err := alltransports.ParseImageName(i.opts.Image)
+	if err != nil {
+		// try adding "docker://"
+		i.opts.Image = "docker://" + i.opts.Image
+		srcRef, err = alltransports.ParseImageName(i.opts.Image)
+		if err != nil {
+			return nil, "", fmt.Errorf("Invalid source name %s: %v", i.opts.Image, err)
+		}
+	}
+
+	certPath, err := i.certPath(i.opts.Image)
+	if err != nil {
+		return nil, "", err
+	}
+	sourceCtx := &types.SystemContext{
+		DockerAuthConfig: nil,
+		DockerCertPath:   certPath,
+	}
+
+	if i.opts.DstPath, err = createOutputDir(i.opts.DstPath, "image-inspector-"); err != nil {
+		return nil, "", err
+	}
+	destRef, err := alltransports.ParseImageName(fmt.Sprintf("dir://%s", i.opts.DstPath))
```

Comment:
This could avoid the string formatting/parsing roundtrip and use [`containers/image/`]`directory.NewReference(i.opts.DstPath)`. But this works just as well.